ros2_foxglove_bridge/src/message_definition_cache.cpp

```diff
@@ -118,19 +118,41 @@ static std::vector<std::string> split_string(const std::string& str,
   return strings;
 }
 
+static std::string trim_string(std::string& str) {
+  constexpr char whitespaces[] = "\t\n\r ";
+  str.erase(0, str.find_first_not_of(whitespaces));  // trim left
+  str.erase(str.find_last_not_of(whitespaces) + 1);  // trim right
+  return str;
+}
+
 /// @brief Split an action definition into individual goal, result and feedback definitions.
 /// @param action_definition The full action definition as read from a .action file
 /// @return A tuple holding goal, result and feedback definitions
 static std::tuple<std::string, std::string, std::string> split_action_msg_definition(
   const std::string& action_definition) {
-  constexpr char SEP[] = "\n---\n";
+  constexpr char SEP[] = "---";
 
-  const auto definitions = split_string(action_definition, SEP);
+  auto definitions = split_string(action_definition, SEP);
```

Comment:
Now is there any risk that this conflict with a default value, e.g. will `string STR="foo---bar"` break it?